### ISSUE TITLE
Docs: In rule examples, combine multiple expected errors

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#3865]
Some of the docs are a bit ugly when there are more than one expected error on a single line, for example:

![selection_003](https://cloud.githubusercontent.com/assets/4616705/9978883/a1e7261c-5f1b-11e5-95d7-62b1f22ce808.png)

Now that https://github.com/eslint/eslint-plugin-markdown/pull/13 has landed, it's possible to simplify these cases to simply `// 2 errors`:

![selection_004](https://cloud.githubusercontent.com/assets/4616705/9978888/fd40d058-5f1b-11e5-846d-23a4b0698407.png)

It can also be done without the space before the `2`, but I'm not sure which version looks better.  It looks more consistent without a space but violates `spaced-comment`.

![selection_005](https://cloud.githubusercontent.com/assets/4616705/9978891/87674c8a-5f1c-11e5-82ff-2670ea195c5f.png)

The downside to combining errors is that it is less explicit about what errors are expected.

**Comment by nzakas** [comment]
Why aren't they also block comments?

**Comment by IanVS** [comment]
They needed to be line comments for the valid-jsdoc rule, so that's how they were implemented in `eslint-plugin-markdown`.  No reason we couldn't implement block comments as well though.

**Comment by nzakas** [comment]
not sure I understand - JSDoc comments are always block comments, and there's no JSDoc comments in your example. How are these related?

**Comment by IanVS** [comment]
Sorry, I should stop trying to reply from my phone and be more explicit.  The expected errors in [valid-jsdoc](http://eslint.org/docs/rules/valid-jsdoc#rule-details) include one case where there are two errors, and using a block comment is not possible inside the JSDoc:

```js
// missing type for @param and missing @returns
/**                                 // 2 errors
 * A description
 * @param num1 The first number.
 */
function foo(num1) {
    // ...
}
```

**Comment by nzakas** [comment]
Then how about changing the other one to a line comment? Switching back and forth between comment styles looks messy to my eye 

Also, I'm not sure "2 errors" is very useful. It's already hard to tell what the actual error is in each example, and hiding the messages doesn't help that.

What if you put the error messages beneath the code instead of to the right?

**Comment by IanVS** [comment]
> Then how about changing the other one to a line comment?

This is possible. Would you want that change throughout all the docs, or just in cases like this where we are mixing messages and counts? Initially they had to be block comments in order to have multiple per line, but now there's no reason they need to remain block comments. I think it could be a simple regex find/replace.

> What if you put the error messages beneath the code instead of to the right?

Again, do you mean in all cases, or only when there are multiple errors on a line? Throughout the docs, I have tried to keep the errors off to the side far enough that they are visually distinct from the code examples themselves, and to me it makes sense to clearly denote the line the error is expected to occur on (by putting the error message on that line). However, I like the idea of including the errors underneath the error count to be explicit. That would keep one error per line for readability, while accommodating the linting. I can mock up an example later tonight to see how it looks.

**Comment by nzakas** [comment]
> Would you want that change throughout all the docs, or just in cases like this where we are mixing messages and counts? 

Throughout. Consistency in documentation is important.

> Again, do you mean in all cases, or only when there are multiple errors on a line? 

Throughout. Consistency in documentation is important. :)

**Comment by IanVS** [comment]
I agree to the first point, not so sure about the second.  I think this would still be consistent:

```js
/*eslint array-bracket-spacing: [2, "never"]*/

var arr = [ 'foo', 'bar' ];   // 2 expected errors:
                              // error: There should be no space after '['
                              // error: There should be no space before ']'

var arr = ['foo', 'bar' ];    // error: There should be no space before ']'

var arr = [ ['foo'], 'bar'];  // error: There should be no space after '['

var arr = [[ 'foo' ], 'bar']; // 2 expected errors:
                              // error: There should be no space after '['
                              // error: There should be no space before ']'
.
.
.
```

The alternative which I think you are suggesting is:

```js
/*eslint array-bracket-spacing: [2, "never"]*/

var arr = [ 'foo', 'bar' ];   
// 2 expected errors:
// error: There should be no space after '['
// error: There should be no space before ']'

var arr = ['foo', 'bar' ];   
// error: There should be no space before ']'

var arr = [ ['foo'], 'bar'];  
// error: There should be no space after '['

var arr = [[ 'foo' ], 'bar'];
// 2 expected errors:
// error: There should be no space after '['
// error: There should be no space before ']'
.
.
.
```

Which in this case doesn't look too bad, but consider this from `curly`:

```js
if (foo) bar();                        
// error: Unnecessary { after 'else'.
else {
    foo++;
}
```

In this case, the error is on line 1, but it's not clear from the example unless you know that the error comment occurs on the line following the error.

Or this from `spaced-comment`:

```js
/* eslint spaced-comment: [2, "always", { "block": { "exceptions": ["-"] } }] */

//--------------    
// error: Expected space or tab after // in comment.
// Comment block
//--------------    
// error: Expected space or tab after // in comment.
```

Again, I would argue this is less readable than the alternative:

```js
/* eslint spaced-comment: [2, "always", { "block": { "exceptions": ["-"] } }] */

//--------------             // error: Expected space or tab after // in comment.
// Comment block
//--------------             // error: Expected space or tab after // in comment.
```

So, how does the example at the very top look to you?

**Comment by nzakas** [comment]
Ugh, I'm really not looking any of this. I'm getting concerned about the complexity of the docs and I'm wondering if we aren't just better off not having the messages in the code samples. Remember, we need people submitting rules to include documentation, and I'm starting to think this is way too much to ask people to do for little benefit. 

I think there's value to linting the code examples, I'm just having serious doubts about how far this is going now, and I think the current conundrum is highlighting that this may not be reasonable.

So I'd like to step back and ask a question: are having the errors inline to the code examples valuable?

**Comment by gyandeeps** [comment]
Closing this in favor of https://github.com/eslint/eslint/issues/4104
@IanVS Feel free to reopen if this is still valid.
